crates/cli/src/parse/timestamp.rs

```diff
@@ -0,0 +1,47 @@
+use ethers::prelude::*;
+use polars::prelude::*;
+use cryo_freeze::{BlockChunk, Fetcher, ParseError, ChunkData, CollectError};
+
+async fn timestamp_to_block_number<P: JsonRpcClient>(
+    timestamp: u64,
+    fetcher: &Fetcher<P>
+) -> Result<u64, CollectError> {
+    let mut block_numbers: Vec<u64> = Vec::new();
+    let latest_block_number = get_latest_block_number(fetcher).await?;
+
+    for block_number in 0..latest_block_number {
+        block_numbers.push(block_number);
+    }
+
+    // perform binary search to determine the block number for a given timestamp.
+    // If the exact timestamp is not found, we return the closest block number.
```

Comment:
if there is a block b with timestamp tb, what happens when you do

- `timestamp_to_block_number(tb)`
- `timestamp_to_block_number(tb - 1)`
- `timestamp_to_block_number(tb + 1)`

like does it always find tb or does it find the block closest to timestamp that is always >= or always <= the block? would be helpful to clarify this in a comment

Reply:
That's a good question. It returns the block closest to the timestamp in terms of absolute distance between the timestamp and the block's timestamp.

However, there exists an edge case pictured below where the timestamp given is equally between two different timestamps.
![image](https://github.com/paradigmxyz/cryo/assets/68152521/56e3dd5a-edea-4421-90e7-254cf19daa59)

I'm currently working on forcing it to return the lower block that has the timestamp <= the inputted timestamp.